com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/HDShadowAlgorithms.hlsl

```diff
@@ -154,6 +154,11 @@ float EvalShadow_PunctualDepth(HDShadowData sd, Texture2D tex, SamplerComparison
 //
 float EvalShadow_AreaDepth(HDShadowData sd, Texture2D tex, float2 positionSS, float3 positionWS, float3 normalWS, float3 L, float L_dist, bool perspective)
 {
+    // TODO: implement less messily. Also probably shouldn't even rely on a global quality setting, but on a per-light shadow algorithm switch.
+#ifdef SHADOW_HIGH // Do PCSS
+    return EvalShadow_PunctualDepth(sd, tex, s_linear_clamp_compare_sampler, positionSS, positionWS, normalWS, L, L_dist, perspective);
```

Comment:
So if we do this, we should move the definition of the algorithm used in the HDShadowAlgorithms.hlsl and do same kind of thing that we do for the other algorithm, so it is configurable

#ifdef SHADOW_ULTRA_LOW
#define PUNCTUAL_FILTER_ALGORITHM(sd, posSS, posTC, tex, samp, bias) SampleShadow_PCF_Tent_3x3(sd.isInCachedAtlas ? _CachedShadowAtlasSize.zwxy : _ShadowAtlasSize.zwxy, posTC, tex, samp, bias)
#define DIRECTIONAL_FILTER_ALGORITHM(sd, posSS, posTC, tex, samp, bias) SampleShadow_Gather_PCF(_CascadeShadowAtlasSize.zwxy, posTC, tex, samp, bias)
#elif defined(SHADOW_LOW)
#define PUNCTUAL_FILTER_ALGORITHM(sd, posSS, posTC, tex, samp, bias) SampleShadow_PCF_Tent_3x3(sd.isInCachedAtlas ? _CachedShadowAtlasSize.zwxy : _ShadowAtlasSize.zwxy, posTC, tex, samp, bias)
...
#endif

Reply:
>but on a per-light shadow algorithm switch.
That's will be an issue.
For this, either you add new variant (and combination will explode) or you do a dynamic branch (and VGPR will explode for area lights (PCSS used way more VGPR than EVSM). 

So for now we will keep it as global.

Reply:
Yup, it just needs a bit of a refactor, since the entire EvalShadow_AreaDepth() needs to reuse what happens inside EvalShadow_PunctualDepth(). In other words it's not sufficient to define the algo for area lights at the same level as PUNCTUAL_FILTER_ALGORITHM

Reply:
> So if we do this, we should move the definition of the algorithm used in the HDShadowAlgorithms.hlsl and do same kind of thing that we do for the other algorithm, so it is configurable

Yup, it just needs a bit of a refactor, since the entire EvalShadow_AreaDepth() needs to reuse what happens inside EvalShadow_PunctualDepth(). In other words it's not sufficient to define the algo for area lights at the same level as PUNCTUAL_FILTER_ALGORITHM